### PR TITLE
wip: Add endpoint errors

### PR DESCRIPTION
**Pull request by mpritham**
## After this PR

==COMMIT_MSG==
==COMMIT_MSG==

### CODE REVIEW

File: conjure-api/src/main/conjure/conjure-api.yml [hunk 1]
```diff
@@ -57,6 +57,11 @@ types:
           - CUSTOM_SERVER
       ErrorDefinition:
         fields:
+          # TODO(pm): Should there be a message here? The docs aren't the same thing: devs can add more context to them,
```
Comment (carterkozak) [inline conjure-api/src/main/conjure/conjure-api.yml:60]: Conjure errors don't provide messages by design, the name is meant to be a key such that a FE component can build a human-readable message. Providing a user-facing message would require our servers to implement internationalization/translation, which is something our current architecture is designed to handle exclusively on the frontend.
While the server can describe what failed, the FE is better positioned to contextualize the failure in order to provide good user experience.

File: conjure-api/src/main/conjure/conjure-api.yml [hunk 2]
```diff
@@ -172,7 +177,18 @@ types:
...
+          # responses:
+          #   union:
+          #     - success: optional<Type>
+          #     - errors: set<TypeName>
           returns: optional<Type>
+          # TODO(pm): put up commit where ErrorName is alias for TypeName.
+          # TODO(pm): Maybe this should actually be `TypeName | ErrorDefinition` to allow devs to create new errors
+          #  inline. Think about if we want to disallow this behavior.
+          #  Let's not allow error types to be imported from other languages. Only Conjure-imports allowed.
+          errors: set<TypeName>
```
Comment (mpritham) [inline conjure-api/src/main/conjure/conjure-api.yml:191]: Do we want to associate each of the `errors` with some documentation? We do this with `returns` already.

File: conjure-api/src/main/conjure/conjure-api.yml (hunk 1 above)
Comment (mpritham) [reply conjure-api/src/main/conjure/conjure-api.yml:60]: Hadn't thought about i18n. Makes sense.

**Comment by mpritham** [comment]
Closing this in favor of another PR introducing similar changes.

**Pull request closed without merging**